### PR TITLE
Docs: Updates docs for redux framework

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Updates documentation for our Redux framework and helpers.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: contribute/style-guides/redux.md [hunk 1]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
 
-## New test functionality
+## Test functionality
 
 ### reducerTester
 
 Fluent API that simplifies the testing of reducers
 
-#### Example
+#### Simple Example
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]:
```suggestion
#### Simple example
```

[Write headings in sentence case, not title case](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#headings)

File: contribute/style-guides/redux.md [hunk 2]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
 
 ```typescript
 reducerTester()
   .givenReducer(someReducer, initialState)
   .whenActionIsDispatched(someAction('reducer tests'))
   .thenStateShouldEqual({ ...initialState, data: 'reducer tests' });
 ```
+
+#### Complex Example
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]:
```suggestion
#### Complex example
```

[Write headings in sentence case, not title case](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#headings)

File: contribute/style-guides/redux.md [hunk 3]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
+Sometimes you encounter a `resulting state` that contains properties that are hard to compare like `Dates` but you still 
+want to compare that other props in state are correct.
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]: This line seems to be hard wrapped. It should be on the same line as the one above.

File: contribute/style-guides/redux.md [hunk 4]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
+
+Then you can use `thenStatePredicateShouldEqual` function on `reducerTester` that will return the `resulting state` that
+you can expect upon.
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]: This line seems to be hard wrapped. It should be on the same line as the one above.

File: contribute/style-guides/redux.md [hunk 5]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
 
 ```typescript
 reducerTester()
   .givenReducer(someReducer, initialState)
   .whenActionIsDispatched(someAction('reducer tests'))
   .thenStateShouldEqual({ ...initialState, data: 'reducer tests' });
 ```
+
+#### Complex Example
+Sometimes you encounter a `resulting state` that contains properties that are hard to compare like `Dates` but you still
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]:
```suggestion
Sometimes you encounter a `resulting state` that contains properties that are hard to compare, such as `Dates`, but you still 
```
"compare like dates" sounds like you want to compare the properties _as_ Dates, which I don't think is the intention here?

File: contribute/style-guides/redux.md [hunk 6]
```diff
@@ -117,17 +117,48 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
+  .whenActionIsDispatched(someAction('reducer tests'))
+  .thenStatePredicateShouldEqual(resultingState => {
+    expect(resultingState.data).toEqual('reducer tests');
+    return true;  
+  });
+```
+
+### thunkTester
+
+Fluent API that simplifies the testing of thunks
```
Comment (marcusolsson) [inline contribute/style-guides/redux.md]:
```suggestion
Fluent API that simplifies the testing of thunks.
```

**Review by marcusolsson — CHANGES_REQUESTED** [review]

File: contribute/style-guides/redux.md (hunk 3 above)
Comment (hugohaggmark) [reply contribute/style-guides/redux.md]: Will fix that!

File: contribute/style-guides/redux.md (hunk 4 above)
Comment (hugohaggmark) [reply contribute/style-guides/redux.md]: Will fix that!

File: contribute/style-guides/redux.md (hunk 1 above)
Comment (dprokop) [inline contribute/style-guides/redux.md]:
```suggestion
#### Usage
```

Comment (hugohaggmark) [reply contribute/style-guides/redux.md]: I like this better, thanks

Comment (dprokop) [reply contribute/style-guides/redux.md]: I would go with `Usage` everywhere when `[Simple|Complex] example` is used

**Review by dprokop — APPROVED** [review]
Good stuff

**Review by marcusolsson — APPROVED** [review]

File: contribute/style-guides/redux.md [hunk 7]
```diff
@@ -117,17 +117,46 @@ Type '{ nonExistingProperty: string; data: string[]; }' is not assignable to typ
...
 reducerTester()
   .givenReducer(someReducer, initialState)
   .whenActionIsDispatched(someAction('reducer tests'))
   .thenStateShouldEqual({ ...initialState, data: 'reducer tests' });
 ```
+
+#### Complex usage
+Sometimes you encounter a `resulting state` that contains properties that are hard to compare, such as `Dates`, but you still want to compare that other props in state are correct.
+
+Then you can use `thenStatePredicateShouldEqual` function on `reducerTester` that will return the `resulting state` thatyou can expect upon.
```
Comment (davkal) [inline contribute/style-guides/redux.md]: s/thatyou/that you/

Comment (davkal) [reply contribute/style-guides/redux.md]: ...expect upon individual properties.

**Review by davkal — APPROVED** [review]
👍

File: contribute/style-guides/redux.md (hunk 7 above)
Comment (hugohaggmark) [reply contribute/style-guides/redux.md]: Thanks David!

**Pull request merged**